pkg/collector/tcp.go

```diff
@@ -86,20 +86,20 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 				if err != nil {
 					klog.Error(err)
 					cp.deleteClient(address)
-					return
+					break
```

Comment:
If we are deleting the client, shouldn't we exit the handTCPclient function?

Reply:
yes you are right, we should not delete client here. Updated.

Reply:
If length cannot be retrieved, then shouldn't we empty the buffer before reading on the connection again.